github_connector/models/github_repository_branch.py

```diff
@@ -275,8 +293,78 @@ def _analyze_code(self):
                         branch.name,
                         e,
                     )
+        # github_analysis_rule
+        for branch in self:
+            rule_ids = self.env["github.analysis.rule"].search(
+                [
+                    "|",
+                    (
+                        "id",
+                        "in",
+                        branch.repository_id.organization_id.analysis_rule_ids.mapped(
+                            "analysis_rule_id"
+                        ).ids,
+                    ),
+                    (
+                        "id",
+                        "in",
+                        branch.repository_id.analysis_rule_ids.mapped(
+                            "analysis_rule_id"
+                        ).ids,
+                    ),
+                ]
+            )
+            if rule_ids:
+                for rule_id in rule_ids:
+                    branch._operations_analysis_rule_id(rule_id)
+
         return True
 
+    def _operations_analysis_rule_id(self, analysis_rule_id):
+        """
+        This function allow to override with other addons that need
+        to change this analysis
+        """
+        self.ensure_one()
+        res = {
+            "paths": [],
+            "code": 0,
+            "documentation": 0,
+            "empty": 0,
+            "string": 0,
+        }
+        for match in analysis_rule_id._get_matches(self.local_path):
+            # _logger.info("analyse file %s "% match)
+            res_file = analysis_rule_id._analysis_file(self.local_path + "/" + match)
+            res["paths"].append(res_file["path"])
+            # define values
+            for key in ("code", "documentation", "empty", "string"):
+                res[key] += res_file[key]
+        # Save info
+        self._operations_analysis_rule_id_save_info(analysis_rule_id, res)
+
+    def _operations_analysis_rule_id_save_info(self, analysis_rule_id, res):
+        self.ensure_one()
+        # remove all
+        self.env["github.repository.branch.rule.info"].search(
```

Comment:
In this function the parameter analysis_rule_id indicate the object (because previously analyzed the code related to this rule), but related to "github.repository.branch.rule.info" (TransientModel) is need to search by organization_id and analysis_rule_id to remove all records previously to create again.